sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -438,11 +438,25 @@ class Analyzer(
             }
             j.copy(right = newRight)
         }
-
       // When resolve `SortOrder`s in Sort based on child, don't report errors as
       // we still have chance to resolve it based on grandchild
-      case s @ Sort(ordering, global, child) if child.resolved && !s.resolved =>
-        val newOrdering = resolveSortOrders(ordering, child, throws = false)
+      case s @ Sort(ordering, global, child) if child.resolved =>
```

Comment:
Should we keep the `&& !s.resolved` in the condition and have another case to handle the case of all literals?